pyomo/contrib/piecewise/tests/models.py

```diff
@@ -0,0 +1,63 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.contrib.piecewise import PiecewiseLinearFunction
+from pyomo.environ import ConcreteModel, Constraint, log, Objective, Var
+
+def make_log_x_model():
+    m = ConcreteModel()
+    m.x = Var(bounds=(1, 10))
+    def log_function(x):
+        return log(x)
+    m.log_function = log_function
+    m.pw_log = PiecewiseLinearFunction(points=[1, 3, 6, 10],
+                                       function=m.log_function)
```

Comment:
Is there a reason this isn't just
```python
m.pw_log = PiecewiseLinearFunction(points=[1, 3, 6, 10], function=math.log)
```

Reply:
The whims of the anti-pickling faction of the coding fairies?

Reply:
That should pickle, though: `math.log` is a global (module scope) function.

Reply:
Yes. You asked for a reason for the original version though. :P